test/functional/eval/capture_spec.lua

```diff
@@ -0,0 +1,72 @@
+local helpers = require('test.functional.helpers')
+local eq = helpers.eq
+local eval = helpers.eval
+local clear = helpers.clear
+local source = helpers.source
+local redir_exec = helpers.redir_exec
+local funcs = helpers.funcs
+
+describe('capture()', function()
+  before_each(clear)
+
+  it('returns the same result with :redir', function()
+    eq(redir_exec('messages'), funcs.capture('messages'))
+  end)
+
+  it('returns the output of the commands if the argument is List', function()
+    eq("foobar", funcs.capture({'echon "foo"', 'echon "bar"'}))
+    eq("\nfoo\nbar", funcs.capture({'echo "foo"', 'echo "bar"'}))
+  end)
+
+  it('supports the nested redirection', function()
+    source([[
+    function! g:Foo()
+      let a = ''
+      redir => a
+      silent echon "foo"
+      redir END
+      return a
+    endfunction
+    function! g:Bar()
+      let a = ''
+      redir => a
+      call g:Foo()
+      redir END
+      return a
+    endfunction
+    ]])
+    eq('foo', funcs.capture('call g:Bar()'))
+
+    exc_exec([[redir => g:mes|echon capture('echo capture("redir => g:mes2|echon 42|redir END")')|redir END]])
+    eq(42, eval('g:mes'))
+    eq(42, eval('g:mes2'))
+  end)
+
+  it('supports the nested echo', function()
+    eq(42, capture('echo capture("echo 42")'))
```

Comment:
I do not see where `capture()` is defined. Did you mean `funcs.capture` or `local capture = funcs.capture` somewhere above?